internal/beater/server_test.go

```diff
@@ -572,6 +576,78 @@ func TestWrapServer(t *testing.T) {
 	require.Equal(t, "true", out["labels"].(map[string]any)["wrapped_reporter"])
 }
 
+func TestWrapServerAPMInstrumentationTimeout(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping slow test")
+	}
+
+	// Enable self instrumentation, simulate a client disconnecting when sending intakev2 request
+	// Check that tracer records the correct http status code
+	found := make(chan struct{})
+	reqCtx, reqCancel := context.WithCancel(context.Background())
+
+	escfg, _ := beatertest.ElasticsearchOutputConfig(t)
+	_ = logp.DevelopmentSetup(logp.ToObserverOutput())
+	srv := beatertest.NewServer(t, beatertest.WithConfig(escfg, agentconfig.MustNewConfigFrom(
+		map[string]interface{}{
+			"instrumentation.enabled": true,
+		})), beatertest.WithWrapServer(
+		func(args beater.ServerParams, runServer beater.RunServerFunc) (beater.ServerParams, beater.RunServerFunc, error) {
+			args.BatchProcessor = modelpb.ProcessBatchFunc(func(ctx context.Context, batch *modelpb.Batch) error {
+				// The service name is set to "1234_service-12a3" in the testData file
+				if len(*batch) > 0 && (*batch)[0].Service.Name == "1234_service-12a3" {
+					// Simulate a client disconnecting by cancelling the context
+					reqCancel()
+					// Wait for the client disconnection to be acknowledged by http server
+					<-ctx.Done()
+					assert.ErrorIs(t, ctx.Err(), context.Canceled)
+					return errors.New("foobar")
+				}
+				for _, i := range *batch {
+					// Perform assertions on the event sent by the apmgorilla tracer
+					if i.Transaction.Name == "POST /intake/v2/events" {
```

Comment:
```suggestion
					if i.Transaction.Id != "" && i.Transaction.Name == "POST /intake/v2/events" {
```

nit: Check for transaction ID to avoid getting an assertion failure if a later span_breakdown event with POST /intake/v2/events comes in.